compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -895,8 +895,12 @@ impl<'a, 'b, 'tcx> BuildReducedGraphVisitor<'a, 'b, 'tcx> {
         self.r.potentially_unused_imports.push(import);
         let imported_binding = self.r.import(binding, import);
         if parent == self.r.graph_root {
-            if let Some(entry) = self.r.extern_prelude.get(&ident.normalize_to_macros_2_0()) {
-                if expansion != LocalExpnId::ROOT && orig_name.is_some() && !entry.is_import() {
+            let ident = ident.normalize_to_macros_2_0();
+            if let Some(entry) = self.r.extern_prelude.get(&ident) {
+                if expansion != LocalExpnId::ROOT
+                    && orig_name.is_some()
+                    && entry.binding.map_or(true, |binding| binding.module().is_some())
```

Comment:
So I made this change: if the binding already exists, and it refers to a module, then we consider it an ambiguity error.

Reply:
> bindings in extern prelude are always modules by definition

There is a special case: it may also refer to `Res::Err` when the external module does not exist.

> we should probably just skip the extern prelude update

I think this is a good solution as well, and I will give it a try.